src/app/core/admin/admin-entity/admin-entity-general-settings/admin-entity-general-settings.component.ts

```diff
@@ -65,18 +92,92 @@ export class AdminEntityGeneralSettingsComponent implements OnInit {
         this.generalSettings.toStringAttributes,
         Validators.required,
       ],
+      hasPII: [this.generalSettings.hasPII],
     });
+    this.showPIIDetails = this.basicSettingsForm.get("hasPII").value;
     this.form = this.fb.group({
       basicSettings: this.basicSettingsForm,
     });
+    this.fetchTableData();
     this.initToStringAttributesOptions();
 
     this.form.valueChanges.subscribe((value) => {
-      // Emit the updated value
       this.generalSettingsChange.emit(this.basicSettingsForm.getRawValue()); // Optionally, emit the initial value
     });
   }
 
+  fetchTableData() {
+    if (this.showPIIDetails) {
+      this.allPIIFields = Array.from(this.entityConstructor.schema.entries())
+        .filter(
+          ([key, field]) =>
+            field.dataType === StringDatatype.dataType && field.label,
+        )
+        .map(([key, field]) => ({
+          key: key,
+          anonymize: field.anonymize,
+          label: field.label,
+        }));
+
+      const data = this.allPIIFields.map((field) => {
+        let anonymize = "Remove";
+        if (field.anonymize === "retain") {
+          anonymize = "Retain";
+        } else if (field.anonymize === "retain-anonymized") {
+          anonymize = "Partially Anonymize";
+        }
+        return { fields: field.label, anonymize: anonymize };
+      });
+
+      this.dataSource = new MatTableDataSource<any>(data);
+    }
+  }
+
+  toggleTable(event: any) {
+    this.showPIIDetails = event.checked;
+    this.basicSettingsForm.get("hasPII").setValue(this.showPIIDetails);
+    this.fetchTableData();
+  }
+  getTooltip(option: string): string {
+    switch (option) {
+      case "Retain":
+        return "Retain: Keep the original value without any anonymization.";
+      case "Partially Anonymize":
+        return "Partially Anonymize: Anonymize the value but retain some information.";
+      case "Remove":
+        return "Remove: Completely remove the value or anonymize it entirely.";
+      default:
+        return "";
+    }
+  }
+
+  selectionOnChange(value: any, selectFielddata: any) {
+    const selectedFieldDetails = this.allPIIFields.find(
+      (field) => field.label === selectFielddata.fields,
+    );
+    const selectedFieldId = selectedFieldDetails.key;
+    const selectedFielddetails =
+      this.entityConstructor.schema.get(selectedFieldId);
+    if (selectFielddata.anonymize == "Partially Anonymize") {
+      selectedFielddetails.anonymize = "retain-anonymized";
+    } else if (selectFielddata.anonymize == "Retain") {
+      selectedFielddetails.anonymize = "retain";
+    } else if ((selectFielddata.anonymize = "Remove")) {
+      selectedFielddetails.anonymize = undefined;
+    }
+    const updatedEntitySchemaMerged = Object.assign(
+      { _isCustomizedField: true },
+      this.entitySchemaField,
+      selectedFielddetails,
+    );
```

Comment:
It should be enough to simply do `this.entityConstructor.schema.get(selectedFieldId).anonymize = newAnonymizeValue` 

also, you can use `<mat-option [value]="..."` to set the exact key (e.g. "retain" instead of "Retain") and avoid having to map it here in the method again.